.travis.yml

```diff
@@ -92,151 +63,149 @@ matrix:
           stage: Initial tests
-          env: MAIN_CMD="flake8 astropy --count $FLAKE8_OPT" SETUP_CMD=''
+          env: TOXENV=codestyle
 
         # Try developer version of Numpy with optional dependencies and also
         # run all remote tests. Since both cases will be potentially
         # unstable, we combine them into a single unstable build that we can
         # mark as an allowed failure below.
-        - os: linux
+        - language: python
+          python: 3.7
           stage: Final tests
-          env: PYTHON_VERSION=3.7 NUMPY_VERSION=dev SETUP_CMD='test --remote-data'
-               CONDA_DEPENDENCIES=''
-               PIP_DEPENDENCIES="$DEV_PIP_DEP $ASDF_DEV"
-               MATPLOTLIB_VERSION=dev
+          env: TOXENV="test-devdeps"
+               TOXPOSARGS="--remote-data=any"
 
         # We check numpy-dev also in a job that only runs from cron, so that
         # we can spot issues sooner. We do not use remote data here, since
         # that gives too many false positives due to URL timeouts.
         # We also install all dependencies via pip here so we pick up the latest
         # releases.
-        - os: linux
-          stage: Cron tests
-          env: NUMPY_VERSION=dev MATPLOTLIB_VERSION=dev EVENT_TYPE='cron'
-               CONDA_DEPENDENCIES=''
-               PIP_DEPENDENCIES="$DEV_PIP_DEP $ASDF_DEV"
+        - language: python
+          python: 3.7
+          stage: Final tests
+          env: TOXENV="test-devdeps"
+               EVENT_TYPE='cron'
 
         # Run documentation link check in a cron job.
         # Was originally in CircleCI doc build but links are too flaky, so
         # we moved it here instead.
-        # NOTE: matplotlib pinning should be consistent with CircleCI HTML job.
-        - os: linux
+        - language: python
+          python: 3.7
           stage: Cron tests
-          env: SETUP_CMD='build_docs -b linkcheck' EVENT_TYPE='cron'
-               MATPLOTLIB_VERSION="<3.1"
-               PIP_DEPENDENCIES="" CONDA_DEPENDENCIES=""
-               INSTALL_WITH_PIP=True
-               EXTRAS_INSTALL="docs"
+          env: TOXENV='linkcheck'
+               EVENT_TYPE='cron'
+          addons:
+              apt:
+                  packages:
+                      - graphviz
+
+        # Testing with Travis provided Python3.8.
+        - language: python
+          stage: Final tests
+          dist: xenial
+          python: 3.8
 
     allow_failures:
-      - os: linux
-        stage: Final tests
-        env: PYTHON_VERSION=3.7 NUMPY_VERSION=dev SETUP_CMD='test --remote-data'
-             CONDA_DEPENDENCIES=''
-             PIP_DEPENDENCIES="$DEV_PIP_DEP $ASDF_DEV"
-             MATPLOTLIB_VERSION=dev
+        - language: python
+          python: 3.7
+          stage: Final tests
+          env: TOXENV="test-devdeps"
+               TOXPOSARGS="--remote-data=any"
 
 before_install:
 
+    # We need a full clone to make sure setuptools_scm works properly
```

Comment:
For another PR, but it would seem that logically `setuptools_scm` would need to know only the last tag that this branch is on...

Reply:
I agree - and indeed probably easiest handled in a separate PR.